packages/grafana-data/src/dataframe/frameComparisons.ts

```diff
@@ -35,16 +35,21 @@ export function compareDataFrameStructures(a: DataFrame, b: DataFrame, skipConfi
       return false;
     }
 
-    // Do not check the config fields
-    if (skipConfig) {
-      continue;
+    // Check if only one field has labels
+    if ((fA.labels != null && fB.labels == null) || (fB.labels != null && fA.labels == null)) {
+      return false;
     }
 
     // Check if labels are different
     if (fA.labels && fB.labels && !shallowCompare(fA.labels, fB.labels)) {
       return false;
     }
```

Comment:
```suggestion
    if (fA.labels || fB.labels) {
      if (fA.labels == null || fB.labels == null) {
        return false; // only one field has labels
      }
      if(!shallowCompare(fA.labels, fB.labels)){
        return false;
      }
    }
```

maybe :man_shrugging: not sure it is any better